bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -231,16 +232,19 @@ public void changeStateTo(Item item, State state) {
                 event.valuecolor = itemUIRegistry.getValueColor(w);
                 event.widgetId = itemUIRegistry.getWidgetId(w);
                 event.visibility = itemUIRegistry.getVisiblity(w);
-                // event.item contains data from the item including its state (in event.item.state)
+                // event.item contains the (potentially changed) data of the item belonging to
+                // the widget including its state (in event.item.state)
+                final Item itemToBeSent = itemBelongsToWidget ? item : getItemForWidget(w);
                 String widgetTypeName = w.eClass().getInstanceTypeName()
                         .substring(w.eClass().getInstanceTypeName().lastIndexOf(".") + 1);
                 boolean drillDown = "mapview".equalsIgnoreCase(widgetTypeName);
                 Predicate<Item> itemFilter = (i -> i.getType().equals(CoreItemFactory.LOCATION));
-                event.item = EnrichedItemDTOMapper.map(item, drillDown, itemFilter, null, null);
+                event.item = EnrichedItemDTOMapper.map(itemToBeSent, drillDown, itemFilter, null, null);
 
-                // event.state is an adjustment of the item state to the widget type.
-                event.state = itemUIRegistry.convertState(w, item, state).toFullString();
-                // In case this state is identical to the item state, its value is set to null.
+                // event.state is an adjustment of the item state to the widget type, but is only sent if both
+                // - the new state belongs to the widget in question and
+                // - the state actually changed
+                event.state = itemBelongsToWidget ? itemUIRegistry.convertState(w, item, state).toFullString() : null;
```

Comment:
Why not rather
`event.state = itemUIRegistry.convertState(w, itemToBeSent, itemBelongsToWidget ? state : event.item.state).toFullString() `

Reply:
Because the lines right below this line (248-250) would set it to null again anyway.

Reply:
No that is wrong, that is all the interest of this field, having in certain cases a widget state that is different from the item state. Why not recomputing it. In case it is equivalent to the item state, yes it will then be set to null.

Reply:
https://github.com/openhab/openhab-core/pull/691/commits/df2284a89b656f33ebb17255167b3e5125155f88 should do what you mean, I guess?

Reply:
Yes that is better like that.
Except the updated comment that is wrong